src/workerd/jsg/function.h

```diff
@@ -231,12 +236,18 @@ class Function<Ret(Args...)> {
     __builtin_unreachable();
   }
 
+  inline void setReceiver(Value receiver) {
```

Comment:
This is my ignorance, but is it legitimate for instances of non-Object types to feature as a receiver in JS?

Reply:
Yep. When a primitive is passed, it will be mapped to the the boxed wrapper.

```
function foo() { console.log(this); }
foo.call(123); // prints `[Number 123]`, this will be a NumberObject
foo.call('hello'); // prints `[String "hello"]`, this will be a StringObject
```